mydemoproject/demoapp1/templates/demoapp1/books.html

```diff
@@ -0,0 +1,67 @@
+{% extends "basic.html" %}
+
+{% block content %}
+<head>
+    <title>Books Page</title>
+</head>
+<section>
+    <h2>Manage Books</h2>
+
+    <!-- Display messages -->
+    {% if messages %}
+        <div class="messages">
+            {% for message in messages %}
+                <div>
+                    {{ message }}
+                </div>
+            {% endfor %}
+        </div>
+    {% endif %}
+
+    <form method="post" action="{% url 'add_book' %}">
+        {% csrf_token %}
```

Comment:
What is the purpose of `csrf_token`?

Reply:
to prevent our django app from Cross request site frogery, it is a cyber security concept.

Reply:
yes it makes sense, CSRF attack is related to attacking innocent authenticated users to click a link through which their token is stolen and attackers can access the information while behaving as an authenticated user.